libraries/standard/mqtt/src/mqtt_lightweight.c

```diff
@@ -1949,6 +1949,16 @@ MQTTStatus_t MQTT_DeserializeAck( const MQTTPacketInfo_t * const pIncomingPacket
                           pSessionPresent );
         status = MQTTBadParameter;
     }
+    /* Pointer for packet identifier cannot be NULL for packets other than
+     * CONNACK and PINGRESP. */
+    else if( ( pPacketId == NULL ) &&
```

Comment:
Do we want to allow args to be `NULL`? Also, if we do this, then we might as well allow `pSessionPresent` to be `NULL` for everything except CONNACK

Reply:
I think yes. It can definitely save some memory from the calling function if they don't need those params. You are right about pSessionPresent. I will add that to the PR.